stackslib/src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -1031,10 +1043,12 @@ impl LeaderBlockCommitOp {
             return Err(op_error::BlockCommitNoParent);
         } else if self.parent_block_ptr != 0 || self.parent_vtxindex != 0 {
             // not building off of genesis, so the parent block must exist
+            // unless the parent is a shadow block
             let has_parent = tx
                 .get_block_commit_parent(parent_block_height, self.parent_vtxindex.into(), &tx_tip)?
                 .is_some();
-            if !has_parent {
+            let maybe_shadow_block = self.parent_vtxindex == 0 && epoch_id.supports_shadow_blocks();
```

Comment:
Oh woops, I meant should this line also have `&& self.parent_block_ptr != 0`

Reply:
I don't think so, because this executes inside of an if-statement whose predicate is `self.parent_block_ptr != 0 || self.parent_vtxindex != 0`.  If `maybe_shadow_block` is `true`, then `self.parent_vtxindex == 0`, so `self.parent_block_ptr != 0`.